blocks/library/gallery/index.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { registerBlockType, query as hpq } from '../../api';
+
+import Placeholder from 'components/placeholder';
+import MediaUploadButton from '../../media-upload-button';
+
+import GalleryImage from './gallery-image';
+
+const { query, attr } = hpq;
+
+const editMediaLibrary = ( attributes, setAttributes ) => {
+	const frameConfig = {
+		frame: 'post',
+		title: wp.i18n.__( 'Update Gallery media' ),
+		button: {
+			text: wp.i18n.__( 'Select' ),
+		},
+		multiple: true,
+		state: 'gallery-edit',
+		selection: new wp.media.model.Selection( attributes.images, { multiple: true } ),
+	};
+
+	const editFrame = wp.media( frameConfig );
+	function updateFn() {
+		setAttributes( {
+			images: this.frame.state().attributes.library.models.map( ( a ) => {
+				return a.attributes;
+			} ),
+		} );
+	}
+
+	editFrame.on( 'insert', updateFn );
```

Comment:
Do we need to detach these event handlers after the edit frame is closed, to avoid risk of zombie event listeners?